src/main/java/pib/pibexception/PibException.java

```diff
@@ -0,0 +1,29 @@
+package pib.pibexception;
+
+import pib.utility.Ui;
+
+/**
+ * PibException class that represents all the exceptions that can be thrown by the Pib class functionalities.
+ */
+public class PibException extends Exception {
+    private String errorType;
+
+    /**
+     * Public constructor for this class
+     *
+     * @param e error message to be used to call the printError method in Ui class
+     */
+    public PibException(String e) {
+        super(e);
+        this.errorType = e;
+    }
+
+    /**
+     * Calls Ui.printError method to print the error message to user
+     *
+     * @return String containing response to be printed to user
+     */
+    public String print() {
+        return Ui.printError(errorType);
+    }
```

Comment:
I think this might be breaking the encapsulation of PibException as Ui knows the types of errorTypes and has to refer to it to return the errorMessage when it is done this way. Perhaps consider moving the switch statement from Ui to here to pass in only the errorMessage to Ui?